### PR TITLE
Irmaku/update node18

### LINKED ISSUE #188: Remove legacy es5 bundling resources
## Is your feature request related to a problem? Please describe.

This repo still contains resources related to the bundling of an es5 module, local demo Sass, and still supports legacy concepts like non-rem CSS post-processing. 

## Describe the solution you'd like

Remove all es5 resources required for bundling. This includes all script references and npm dependencies. 

Remove these legacy services. Also, make sure that they are removed from all documentation including the README and API docs. 

Update the index.html file in the ./dir to use CDN style resources. The local demo Sass resources should be deleted. The npm script for processing these demo styles should also be released

Please delete any `_legacy` directory if one exists. 

## Exit criteria 

This issue will be complete once all legacy Sass and es5 building resources have been removed from the repo. 

See linked commits for update references. 

> This issue was copied from [#61](https:&#x2F;&#x2F;github.com&#x2F;AlaskaAirlines&#x2F;auro-header&#x2F;issues&#x2F;61) by @blackfalcon.

- https://github.com/AlaskaAirlines/auro-header/pull/84/commits/57ca022c3fc05c9ee245df0d77489f1b66bccd4f
- https://github.com/AlaskaAirlines/auro-header/pull/84/commits/9f4c8f41011b3fa6a748d54296be41f86a5ae9c4

> This issue was copied from [#56](https:&#x2F;&#x2F;github.com&#x2F;AlaskaAirlines&#x2F;auro-flight&#x2F;issues&#x2F;56) by @blackfalcon.

### LINKED ISSUE #189: Update to Lit2.0, TS support, issue templates, component registration extension
## Is your feature request related to a problem? Please describe.

The badge element is built using Lit1.0 and related dependencies. The element does not support typescript users.  Issue templates and github settings in this repo are outdated. The current badge element is well behind recent versions of core support functions. 

This repo does not support TS users. 

This repo's issue templates are out of date. 

This repo does not support component extension. 

## Describe the solution you'd like

**Update to Lit2.0**

Using a Node 18 environment, update all other dependencies. This may include deleting and rebuilding the `package-lock.json` file and `node_modules/` directory then running `npm run i` to recreate all resources. 

**Update component registration**

Update the following core features of the badge element 

1. Remove the element definition from the element .js file and place into index.js file
1. Update rollup dependencies and configs to support the new import model

**Huskey 8** 

The following outlines the steps needed for updating any repo to the latest version of Husky.

1. npx husky-init
1. npm i @open-wc/testing@latest @web/test-runner@latest @web/dev-server@latest --save-dev
1. npm run prepare
1. chmod ug+x .husky/*
1. chmod ug+x .git/hooks/*
1. Update package.json - "test": "wtr --coverage", and "test:watch": "wtr --watch",
1. Add web-test-runner.config.mjs
1. Update .husky/pre-commit and necessary pre-commit actions
1. Delete legacy Husky configs from package.json
1. Delete karma.conf.js file
1. Uninstall @open-wc/testing-karma

**Update typescript support** 

Update the repo to support typescript. The solution is to use the `typescript` npm package, add a `types` npm build script and create a `tsconfig.json` file. 

#### tsconfig.json example 

```
{
  "include": ["src/**/*.js"], // process the generated JS files from the src js files 
  "exclude": [],
  "compilerOptions": {
    // Tells TypeScript to read JS files, as
    // normally they are ignored as source files
    "allowJs": true,
    // Generate d.ts files
    "declaration": true,
    // This compiler run should
    // only output d.ts files
    "emitDeclarationOnly": true,
    "outDir": "dist",
    // go to js file when using IDE functions like
    // "Go to Definition" in VSCode
    "declarationMap": true
  }
}
```

**Update the repo's ISSUE TEMPLATES and add CODEOWNER file** 

1. update:   .github/ISSUE_TEMPLATE/bug_report.md
1. update:   .github/ISSUE_TEMPLATE/feature_request.md
1. update:   .github/ISSUE_TEMPLATE/general-support.md
1. update:   .github/settings.yml
1. delete:    .github/ISSUE_TEMPLATE/develop-design-story.md
1. delete:    .github/ISSUE_TEMPLATE/audit.md
1. delete:    .github/ISSUE_TEMPLATE/new_wc.md
1. delete:    .github/ISSUE_TEMPLATE/parent-design-story.md
1. delete:    .github/ISSUE_TEMPLATE/review-design-story.md
1. delete:    .github/ISSUE_TEMPLATE/symbol-design-story.md
1. delete:    .github/ISSUE_TEMPLATE/user-story.md
1. new file:   .github/CODEOWNERS

When reviewing code from the previous issue commit, be sure to include this commit as well. 
https://github.com/AlaskaAirlines/auro-badge/commit/fdaddcc8d314decfa891a1930ba90d31293943f2

**Update to support Node 18**

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/package.json#L18-L20

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/.github/workflows/testPublish.yml#L18

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/.github/workflows/testPublish.yml#L40

**Dart Sass**

Update project and npm scripts to use the latest version of Dart Sass

**Single CLI**

Allow users to use a single CLI instance for development

**Update to use Chalk 5**

Update package and postinstall to support Chalk 5+

**@aurodesignsystem**

Update any current dependency and code resources to support any new element published to the `@aurodesignsystem` namespace. 

## Exit criteria 

This issue will be considered closed once the repo is updated to Lit2.0 and all features and tests operate as expected with all other dependencies updated as well. 

This issue can be closed once a .d.ts file is created with the build. 

This issue can be closed once the issue templates match the content from the generator repo. 

This issue will not be considered complete once the listed core features are updated along with all associated dependencies and all features work as previously expected with node 18 installed.

### LINKED ISSUE #190: Update to Node 18
## Is your feature request related to a problem? Please describe.

Repo is structured to only support Node 16. 

## Describe the solution you'd like

Update to support Node 18.

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/package.json#L18-L20

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/.github/workflows/testPublish.yml#L18

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/.github/workflows/testPublish.yml#L40

Update all action dependencies to `v3`

https://github.com/AlaskaAirlines/auro-flight/blob/50854e9213f571c6137303fa466a7ea9a3df944a/.github/workflows/testPublish.yml#L21-L43

## Exit criteria 

This issue is complete once all the code references have been updated to support Node 18 and the latest dependency versions. 

> This issue was copied from [#58](https:&#x2F;&#x2F;github.com&#x2F;AlaskaAirlines&#x2F;auro-flight&#x2F;issues&#x2F;58) by @blackfalcon.

### LINKED ISSUE #191: Update legacy dependencies
# General Support Request

A series of dependencies have become out of date. 

## Support request

Update all dependency versions. 

## Possible Solution

This is a great tool to selectively update

https://www.npmjs.com/package/npm-check

## Exit criteria 

This issue is complete once all the local dependencies are updated the component rebuilds as expected. 

> This issue was copied from [#59](https:&#x2F;&#x2F;github.com&#x2F;AlaskaAirlines&#x2F;auro-flight&#x2F;issues&#x2F;59) by @blackfalcon.

> This issue was copied from [#63](https:&#x2F;&#x2F;github.com&#x2F;AlaskaAirlines&#x2F;auro-flight&#x2F;issues&#x2F;63) by @blackfalcon.

### LINKED ISSUE #193: Update .npmignore
# General Support Request

With new repo structure, .npmignore needs to be updated. 

## Support request

```
# standard ignore
*config*
*conf*
LICENSE
NOTICE

.github/
.gitignore
.eslintrc
.stylelintrc

coverage/
docs/
scripts/
test/
apiExamples/
docTemplates/
demo/**/*.html

dist/auro-*.js
!dist/auro-*__bundled.js
dist/style-css.js

# ignore src
src/**/*.scss
src/**/*.css
```

## Exit criteria 

This issue is closed once the .npmignroe file matches the proposed spec.

### LINKED ISSUE #194: Update .gitignore
# General Support Request

Update .gitignore to address new repo structure 

## Support request

```
.DS_Store
node_modules/
docTemplates/
.cache/

## Demo support files
demo/*.css
demo/*.css.map

## CSS built from Sass process
src/*.css
src/*.css.map

## *-css built from sass-render process
*-css.js

## dist files
dist/
build/

## Code coverage reports
coverage/
```

## Exit criteria 

This issue is complete once the .gitignore meets the provided spec.

### PR DESCRIPTION
**Pull request by irma-kurnia-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Resolves: #188 , #189 , #190 , #191 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    attributes:
+      value: >
+        [![See it on
+        NPM!](https://img.shields.io/npm/v/@alaskaairux/auro-dropdown?style=for-the-badge&color=orange)](https://www.npmjs.com/package/@alaskaairux/auro-dropdown)
+  - type: textarea
+    id: details
+    attributes:
+      label: Please describe the bug
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
      description: List out the steps to reproduce the behavior and include screenshots
      placeholder: |
        1. Go to '...'
        2. Click on '....'
        3. Scroll down to '....'
        4. See error
```

File: package.json [hunk 2]
```diff
@@ -16,71 +16,65 @@
...
   },
   "dependencies": {
-    "@popperjs/core": "^2.11.6",
-    "chalk": "^4.1.2",
-    "lit-element": "^2.5.1",
-    "lit-html": "^1.4.1"
+    "@popperjs/core": "^2.11.7",
+    "chalk": "^5.2.0",
+    "lit": "^2.7.4",
+    "lit-html": "^2.7.3"
```
Comment (blackfalcon) [inline package.json]:
```suggestion

```

Uninstall this dependency, this is no longer needed with Lit2.0

File: package.json [hunk 3]
```diff
@@ -16,71 +16,65 @@
...
-    "@semantic-release/npm": "^9.0.0",
-    "@web/dev-server": "^0.1.35",
-    "@web/test-runner": "^0.15.0",
-    "autoprefixer": "^10.4.13",
-    "babel-loader": "^9.1.0",
+    "@semantic-release/npm": "^10.0.3",
+    "@web/dev-server": "^0.2.1",
+    "@web/test-runner": "^0.16.1",
+    "autoprefixer": "^10.4.14",
+    "babel-loader": "^9.1.2",
```
Comment (blackfalcon) [inline package.json]: Remove dependency

File: package.json [hunk 4]
```diff
@@ -16,71 +16,65 @@
...
-    "husky": "^8.0.2",
+    "core-js": "^3.30.1",
+    "eslint": "^8.39.0",
+    "eslint-plugin-jsdoc": "^43.1.1",
+    "husky": "^8.0.3",
     "lodash": "^4.17.21",
     "markdown-magic": "^2.6.1",
-    "marked": "^4.2.5",
-    "nodemon": "^2.0.20",
+    "marked": "^5.0.0",
```
Comment (blackfalcon) [inline package.json]: Remove dependency

File: package.json [hunk 5]
```diff
@@ -16,71 +16,65 @@
...
     "npm-run-all": "^4.1.5",
-    "postcss": "^8.4.20",
-    "postcss-custom-properties": "^13.1.3",
-    "postcss-discard-comments": "^5.0.1",
-    "postcss-rem-to-pixel": "^4.1.2",
-    "postcss-remove-rules": "^1.0.0",
+    "postcss": "^8.4.23",
+    "postcss-discard-comments": "^6.0.0",
     "postcss-selector-replace": "^1.0.2",
     "prismjs": "^1.29.0",
```
Comment (blackfalcon) [inline package.json]: Remove dependency

File: package.json [hunk 6]
```diff
@@ -134,37 +128,30 @@
...
-    "build:docs": "npm run build:markdownDocs",
-    "build:sass": "npm-run-all build:sass:demo build:sass:component postCss:component sass:render",
-    "build:sass:demo": "sass --no-source-map demo:demo",
+    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
+    "build:sass": "npm-run-all build:sass:component postCss:component sass:render",
+    "build:sass:dev": "npm-run-all build:sass:component postCss:component sass:render",
+    "build:sass:demo": "sass ./demo/sass/ --output ./demo/css/",
     "build:sass:component": "sass --no-source-map src:src",
-    "build:watch": "nodemon -e scss,js,html --watch src --watch apiExamples --exec npm run build:dev:assets",
+    "build:watch": "nodemon -e scss,js --watch src --exec npm run build:dev:assets",
```
Comment (blackfalcon) [inline package.json]:
```suggestion
    "build:watch": "nodemon -e scss,js,html --watch src --watch apiExamples --exec npm run build:dev:assets",
```

Undo this change. The previous script was correct.

File: package.json [hunk 7]
```diff
@@ -134,37 +128,30 @@
...
     "bundler": "rollup -c",
     "bundler:test": "rollup -c -w",
-    "scssLint": "stylelint \"./src/**/*.scss\"",
-    "demo:copy:demo": "copyfiles -u 1 -V './demo/demo.md' ./build/demo",
-    "demo:copy:index": "copyfiles -u 1 -V './demo/index.html' ./build",
-    "demo:new:build": "mkdir ./build && mkdir ./build/css && mkdir ./build/demo",
-    "demo:rm:build": "rm -rf ./build",
-    "demo:update:index": "node ./scripts/prepForBuild",
+    "cssLint": "stylelint \"./src/*.css\"",
```
Comment (blackfalcon) [inline package.json]: Revert

File: package.json [hunk 8]
```diff
@@ -134,37 +128,30 @@
...
     "dev": "concurrently --kill-others 'npm run build:watch' 'npm run serve'",
     "dist:js": "copyfiles -u 1 -V './src/**/*.js' ./dist",
     "esLint": "./node_modules/.bin/eslint src/**/*.js",
-    "linters": "npm-run-all scssLint esLint",
+    "linters": "npm-run-all cssLint esLint",
```
Comment (blackfalcon) [inline package.json]: Revert

File: package.json [hunk 9]
```diff
@@ -134,37 +128,30 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render dist:js build:api test build:markdownDocs bundler postinstall",
-    "build:ci": "npm-run-all sweep build",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api types",
```
Comment (blackfalcon) [inline package.json]:
```suggestion
    "build": "npm-run-all build:sass sass:render dist:js bundler postinstall build:api types",
```

File: src/style.scss [hunk 10]
```diff
@@ -3,34 +3,27 @@
...
   display: inline-block;
   box-sizing: border-box;
-
   border-width: 1px;
   border-style: solid;
   border-color: transparent;
-
   background-color: var(--auro-color-background-lightest);
-  box-shadow: -2px 0 5px 2px rgba(0 0 0 / .08), 0 2px 5px 1px rgba(0 0 0 / .08);
+  box-shadow: -2px 0 5px 2px rgb(0 0 0 / .08), 0 2px 5px 1px rgb(0 0 0 / .08);
```
Comment (blackfalcon) [inline src/style.scss]: Why was this changed? Please revert.

File: package.json [hunk 11]
```diff
@@ -134,37 +128,30 @@
...
     "build:api": "wca analyze 'src/auro-dropdown.js' --outFiles docs/api.md",
-    "build:demo": "npm-run-all build demo:rm:build demo:new:build demo:copy:index demo:copy:demo demo:update:index",
-    "build:dev:assets": "npm-run-all build:sass:demo build:sass:component postCss:component sass:render build:docs",
-    "build:markdownDocs": "node scripts/generateDocs.js",
```
Comment (blackfalcon) [inline package.json:142]: Revert this removed script and update to the following. 

```
"build:markdownDocs": "node scripts/generateDocs.js",
```

File: .husky/pre-commit [hunk 12]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-./node_modules/.bin/npm-run-all preCommit test linters postinstall
+./node_modules/.bin/npm-run-all linters test
```
Comment (blackfalcon) [inline .husky/pre-commit]:
```suggestion
./node_modules/.bin/npm-run-all preCommit test linters postinstall
```

File: .stylelintrc [hunk 13]
```diff
@@ -1,137 +1,9 @@
...
-        ]
-      }
-    ],
-    "shorthand-property-no-redundant-values": true,
-    "string-quotes": "single",
-    "value-list-comma-newline-before": "never-multi-line"
+    "no-duplicate-selectors": null,
+    "font-family-no-duplicate-names": null,
+    "no-descending-specificity": null,
+    "selector-type-no-unknown": null
```
Comment (blackfalcon) [inline .stylelintrc]: Please revert the whole file. It should be the same as this resource. 

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.stylelintrc

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by irma-kurnia-alaska** [comment]
Hi @blackfalcon, 

I will updated as you mention feedback for this PR. Thank you for your feedback.

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**